src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -8,34 +8,60 @@
 import com.revrobotics.CANSparkMax;
 
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.classes.DebugLog;
+import frc.robot.classes.TimeOfFlightSensor;
 import frc.robot.hardware.MotorController;
 import frc.robot.hardware.MotorController.MotorConfig;
+import frc.robot.subsystems.ArmSubsystem.ArmState;
 import edu.wpi.first.networktables.GenericEntry;
 import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
 import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 public class IntakeSubsystem extends SubsystemBase implements AutoCloseable {
+  public enum FlightStates{
+    IDLE,
+    CUBE,
+    CONE,
+    CONE_SCORE
+  }
+
   public static final double kConeIntakeSpeed = -0.75;
   public static final double kConeOuttakeSpeed = 0.75;
   public static final double kCubeIntakeSpeed = 0.55;
   public static final double kCubeOuttakeSpeed = -0.55;
+  private TimeOfFlightSensor timeOfFlightSensor;
+
+  private final double coneActivationThreshold = 25.0; // placeholder value for how small values have to be for cone to be there
+  private final double cubeActivationThreshold = 25.0; // placeholder value for how small values have to be for cube to be there
+  private FlightStates tofState = FlightStates.IDLE;
 
   private CANSparkMax motor;
   private CANSparkMax motor2;
   private static ShuffleboardTab matchTab = Shuffleboard.getTab("Match");
   private static GenericEntry intakeEntry = matchTab.add("Intake Speed", 0.0).getEntry();
   private static GenericEntry intakeMode = matchTab.add("Intake Mode", "Cone Mode").getEntry();
+
+  private DebugLog coneDist = new DebugLog<Integer>(0, "Cone Distance", this);
+  private DebugLog cubeDist = new DebugLog<Integer>(0, "Cube Distance", this);
+  private DebugLog hasCone = new DebugLog<Boolean>(false, "Has Cone", this);
+  private DebugLog hasCube = new DebugLog<Boolean>(false, "Has Cube", this);
+  private DebugLog isConeSensor = new DebugLog<Boolean>(true, "Cone Sensor Working", this);
+  private DebugLog isCubeSensor = new DebugLog<Boolean>(true, "Cube Sensor Working", this);
+
+
   private boolean isConeMode;
 
   /** Creates a new IntakeSubsystem. */
-  public IntakeSubsystem() {
+  public IntakeSubsystem(TimeOfFlightSensor timeOfFlightSensor) {
     isConeMode = true;
+    this.timeOfFlightSensor = timeOfFlightSensor;
 
     motor = MotorController.constructMotor(MotorConfig.IntakeMotor1);
     motor2 = MotorController.constructMotor(MotorConfig.IntakeMotor2);
   }
   
-  private void spinWheels(double velocity) {
+  public void spinWheels(double velocity) {
```

Comment:
Use pull() and push() instead of spinWheels directly. Make this private again

Reply:
We run the motors at a different (lower) speed when trying to keep in-game pieces. Push and pull don't take in arguments for velocity while spinWheels does

Reply:
Added new hold() method in subsystem